pkg/workflows/sdk/v2/promise.go

```diff
@@ -0,0 +1,58 @@
+package sdk
+
+import (
+	"sync"
+)
+
+type Promise[T any] interface {
+	Await() (T, error)
+	CapabilityPromise
+}
+
+type CapabilityPromise interface {
+	promise()
+}
+
+func NewBasicPromise[T any](await func() (T, error)) Promise[T] {
+	return &basicPromise[T]{await: await}
+}
+
+func PromiseFromResult[T any](result T, err error) Promise[T] {
+	return &basicPromise[T]{resolved: result, err: err, isResolved: true}
+}
+
+type basicPromise[T any] struct {
```

Comment:
From the other draft: Is this different from [sync.OnceValues](https://pkg.go.dev/sync#OnceValues), or could that be used, either here or directly by callers?

Reply:
I don't want to use it directly because we'll likely add await (or awaitN) all at some point and we may need to add some methods to promise to support it. That's also why I made the interface private, so I can add methods and know all implementations have them.

sdk.Await2(...) (T1, T2, err)
sdk.Await3(...)(T1, T2, T3, err)